hexrd/imageseries/save.py

```diff
@@ -331,6 +363,117 @@ def extract_data(i):
         arrd.update(self._process_meta())
         np.savez_compressed(self.cache, **arrd)
 
+    def _write_frames_fch5(self):
+        """Write framecache into an hdf5 file. The file will use three
+        datasets for the framecache:
+        - 'data': (m,1) array holding the datavalues of all frames. `m` is
+          evaluated upon runtime
+        - 'indices': (m,2) array holding the row& col information for the
+          values in data. 'data' together within 'indices' represent tha data
+          using the CSR format for sparse matrices.
+        - 'frame_ids`: (2*nframes)  holds the range that the i-th frame
+          occupies in the above arrays. i.e. the information of the i-th frame
+          can be accessed using:
+
+          data_i = data[frame_ids[2*i]:frame_ids[2*i+1]] and
+          indices_i = indices[frame_ids[2*i]:frame_ids[2*i+1]]
+        """
+        max_frame_size = self._ims.shape[0] * self._ims.shape[1]
+        nframes = len(self._ims)
+        shape = self._ims.shape
+        data_dtype = self._ims.dtype
+
+        frame_indices = np.empty((2 * nframes,), dtype=np.uint64)
+        data_dataset = None
+        indices_dataset = None
+        file_position = 0
+        total_size = 0
+
+        common_lock = threading.Lock()
+        thread_local = threading.local()
+
+        # creating an array in memory will fail if data is too big or threshold
+        # too low, so we write to the file while iterating the frames
+        with h5py.File(self.cache, "w") as h5f:
+            h5f.attrs['HEXRD_FRAMECACHE_VERSION'] = 1
+            h5f["shape"] = shape
+            h5f["nframes"] = nframes
+            h5f["dtype"] = str(self._ims.dtype).encode()
+            metadata = h5f.create_group("metadata")
+            unwrap_dict_to_h5(metadata, self._process_meta())
```

Comment:
```suggestion
            unwrap_dict_to_h5(metadata, self._meta)
```

Can you just do this instead? I think we don't need `self._process_meta()` for the fch5 format.

We should also verify that this saves nested metadata correctly!